add colemak layout #129

dircsem opened the pull request.

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+<keyboard botton_row="false">
```
There's a typo so this has no effect.
```suggestion
<keyboard>
```

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+    <key key0="b" key1="(" key2=")"/>
+    <key key0="k" key1=";" key2=":"/>
+    <key key0="m" key1="|" key2="\?" />
+    <key key0="backspace" shift="0.25" key1="delete" width="1.25"/>
```
Why this empty space between `m` and backspace ?

dircsem replied:
This is because if they are together you can press delete instead of m

Julow replied:
Makes sense but it's the only layout that have this. I wonder if this should be an option available to everyone.
Anyway, I don't mind this.

dircsem replied:
Yes it could be an option, I haven't seen anything related to the app code I`ll try to find out where are the interfaces and try to place an option for that.

Julow replied:
Let's not worry about the option yet as I'm not yet sure it is useful. Let's keep the gap in the layout and see how it is on the long run. Maybe other people will share their opinions.

dircsem replied:
Ok I`ve take a look on how to put this option but it's another issue.
I've based this layout, the character keys, in the picture bellow:
https://lh5.ggpht.com/Y6MxBScWfyORVaMMkgMAO3Kl63ULQXTisNYIjCZdeU2c9hIy_j4bvXZyWoPms_I02_NA=h500
But I didn't followed the layout for symbols and had tried to put it when I thought that it was better.
I think that this ok, I'll keep using it and see if there are any keys that could be changed, please do a review to see if all necessary keys are here.
# this is another issue
Can the accents be merge in to a one key with their characters? Like merging "accent_tilde" with ~ character, if the user press it and press a compatible key like a it became ã, if the user press it and space it became ~.

Julow replied:
> Can the accents be merge in to a one key with their characters?

No, this is not possible.

dircsem replied:
This in fact is exact the behavior, but now we have to add like accent_tilde and ~ keys, but we could add it as one key, the "~" acting as accent_tilde instead of adding it twice.
In my opinion this would make easy to visualize at keyboard and would bring the same behavior as we have at computer keyboards.

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+    <key key0="n" key1="accent_ogonek"/>
+    <key key0="e" key1="accent_circonflexe" key2="accent_aigu" />
+    <key key0="i" key1="\\" key2="/"/>
+    <key key0="o" />
```
The macron accent is missing, `o` is a good place for it:
```suggestion
    <key key0="o" key1="accent_macron" />
```

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+  <row>
+    <key key0="a" key2="tab"/>
+    <key key0="r" key1="accent_grave" key2="accent_tilde"/>
+    <key key0="s" key1="`" key2="~" />
```
Missing key:
```suggestion
    <key key0="s" key1="`" key2="~" key3="ß" />
```
Don't worry, this key is only shown to German users.

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+    <key key0="t" key1="accent_double_aigu" key2="accent_ring"/>
+    <key key0="d" key1="&quot;" key2="'" />
+    <key key0="h"  />
+    <key key0="n" key1="accent_ogonek"/>
```
An other missing accent:
```suggestion
    <key key0="n" key1="accent_ogonek" key2="accent_dot_above"/>
```

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+    <key key0="j" key1="accent_caron" key2="6" key3="^" key4="accent_trema"/>
+    <key key0="l" key2="7" key3="&amp;"/>
+    <key key0="u" key2="8" key3="*"/>
+    <key key0="y" key2="9" />
```
`-` and `_` are missing ! These are pretty important keys. I suggest to put them at the bottom of the `y` key:
```suggestion
    <key key0="y" key2="9" key3="-" key4="_" />
```

Julow commented on res/xml/colemak.xml:
```diff
@@ -0,0 +1,43 @@
+    <key key0="h"  />
+    <key key0="n" key1="accent_ogonek"/>
+    <key key0="e" key1="accent_circonflexe" key2="accent_aigu" />
+    <key key0="i" key1="\\" key2="/"/>
```
`=` and `+` are missing, I suggest to move `\\` and `/` to make room:
```suggestion
    <key key0="i" key1="=" key2="+" key3="\\" key4="/"/>
```

Julow reviewed (commented):
Thanks!

What's version of Colemak is it ? The `,` on the bottom right feels unusual.
There's a few missing keys, I suggested a place to insert them.

dircsem commented:
I've changed the layout on the last few days and I`ve removed on key from the top, also moved numbers to the second line. I`ll merge your requests and see how it'll be.
I've seen the missing =, the other keys I've need to use ;)
![photo_2022-05-07_17-12-40](https://user-images.githubusercontent.com/2041058/167270341-952310e9-e5e9-4f3e-92b0-c1cf573a7637.jpg)
I'll post a picture after merging.

Julow reviewed (approved):
Merging when you think it's ready.

dircsem commented:
I think that you can merge it already, I'll test for another week and if I get any other changes I can push next weekend.

Julow commented:
Merging :) Thanks !

Sorry for the delay, I planned on not releasing on the month of May.
